modules/tinymce/src/core/main/ts/content/SetContent.ts

```diff
@@ -6,8 +6,31 @@
  */
 
 import Editor from '../api/Editor';
+import AstNode from '../api/html/Node';
 import * as Rtc from '../Rtc';
 import { Content, SetContentArgs } from './ContentTypes';
+import { postProcessSetContent, preProcessSetContent } from './PrePostProcess';
 
-export const setContent = (editor: Editor, content: Content, args: Partial<SetContentArgs> = {}): Content =>
-  Rtc.setContent(editor, content, args);
+const defaultFormat = 'html';
+
+const isTreeNode = (content: unknown): content is AstNode =>
+  content instanceof AstNode;
+
+const setupArgs = (args: Partial<SetContentArgs>, content: Content): SetContentArgs => ({
+  format: defaultFormat,
+  ...args,
+  set: true,
+  content: isTreeNode(content) ? '' : content
+});
+
+export const setContent = (editor: Editor, content: Content, args: Partial<SetContentArgs> = {}): Content => {
+  const defaultedArgs = setupArgs(args, content);
+
+  return preProcessSetContent(editor, defaultedArgs).map((updatedArgs) => {
+    // Don't use the content from the args for tree, as it'll be an empty string
+    const updatedContent = isTreeNode(content) ? content : updatedArgs.content;
+    const { content: _content, html } = Rtc.setContent(editor, updatedContent, updatedArgs);
+    postProcessSetContent(editor, html, updatedArgs);
+    return _content;
```

Comment:
Why is this called `_content`? That naming scheme is only meant to be used for unused attributes, but it's definitely used here. Maybe instead of destructuring above, do something like this?
```
    const result = Rtc.setContent(editor, updatedContent, updatedArgs);
    postProcessSetContent(editor, result.html, updatedArgs);
    return result.content;
```

Reply:
Initially, I wanted to avoid using the additional variable name and confusing associations with the` Result` type. But destruction to {content, html} ends up to strange error for the previous line in code :
`error TS2448: Block-scoped variable 'content' used before its declaration.`
 even though the `content` argument looks visible in scope.

Reply:
It's possibly related to the fact that `content` is already in scope. Either way, destructuring isn't really getting any benefit here and I personally don't think it's confusing with the `Result` type. However, if you're worried about that then we should just come up with another name instead... maybe `contentResult` or `setContentResult` to match the `SetContentResult` type?

Reply:
I've reverted to the initially suggested variant without destructuring